Apidaze.SDK.Tests.Integration/XML/Program.cs

```diff
@@ -0,0 +1,201 @@
+            {
+                _keepGoing = false;
+                lock (Listener)
+                {
+                    Listener.Stop();
+                }
+                try
+                {
+                    _mainLoop.Wait();
+                }
+                catch
+                {
+                    // ignored
+                }
+            }
+
+            /// <summary>
+            /// The main loop to handle requests into the HttpListener
+            /// </summary>
+            private static async Task MainLoop()
+            {
+                Listener.Prefixes.Add(LOCALHOST + INTRO_PATH);
+                Listener.Start();
+                while (_keepGoing)
+                {
+                    try
+                    {
+                        var context = await Listener.GetContextAsync();
+                        lock (Listener)
+                        {
+                            if (_keepGoing) ProcessRequest(context);
+                        }
+                    }
+                    catch (Exception e)
+                    {
+                        Console.WriteLine(e.Message);
+                        if (e is HttpListenerException) return;
+                    }
+                }
+            }
+
+            /// <summary>
+            /// Handle an incoming request
+            /// </summary>
+            /// <param name="context">The context of the incoming request</param>
+            private static void ProcessRequest(HttpListenerContext context)
+            {
+                using var response = context.Response;
+                try
+                {
+                    Console.WriteLine(context.Request.RawUrl + "\n" + DateTime.Now);
+                    var handled = context.Request.Url.AbsolutePath switch
+                    {
+                        "/" => GetResponse(context, response),
+                        _ => false
+                    };
+                    if (!handled)
+                    {
+                        response.StatusCode = 404;
+                    }
+
+                    Console.WriteLine(response.ContentType + response.StatusDescription);
+                }
+                catch (Exception e)
+                {
+                    ReturnExceptionResponse(response, e);
+                }
+            }
+
+            /// <summary>
+            /// Gets the response.
+            /// </summary>
+            /// <param name="context">The context.</param>
+            /// <param name="response">The response.</param>
+            /// <returns><c>true</c> if handled, <c>false</c> otherwise.</returns>
+            private static bool GetResponse(HttpListenerContext context, HttpListenerResponse response)
+            {
+                var handled = false;
+                switch (context.Request.HttpMethod)
+                {
+                    case "GET":
+                        var script = ApidazeScript.Build();
```

Comment:
This  `context.Request.HttpMethod` should also be moved into `ProcessRequest`

Reply:
ok